File: Backend/TravelPlanner.Core/DomainModels/DayPlan.cs
using System.Runtime.Serialization;

namespace TravelPlanner.Core.DomainModels
{
    [DataContract]
    public class DayPlan
    {
        [DataMember]
        public string DayPlanId { get; set; }

        [DataMember]
        public Itinerary[] Days { get; set; }
    }
}

File: TravelPlannerApp/app/src/main/java/com/travelplanner/models/DayPlan.kt
package com.travelplanner.models

import java.time.LocalDate
import java.time.LocalDateTime

class DayPlan(val dayPlanId: String, val days: List<Itinerary>)

class Itinerary(val itineraryId: String, val date: LocalDate, val itineraryItems: List<ItineraryItem>)

class ItineraryItem(val itineraryItemId: String, val description: String, val poi: Poi, val title: String)